books/ihs/basic-definitions.lisp

```diff
@@ -454,6 +454,22 @@ is shifted to the left by @('size') bits before the merge.</p>"
        ;; Could probably use logior instead of +.
        :exec (+ (loghead size i) (ash j size))))
 
+(defsection logappn
+  :short "N-ary version of @(tsee logapp)."
+  :long
+  "<p>This supports readable appending of multiple bit chunks, e.g.</p>
+   <code>
+   (logappn 32 base
+            8 padding
+            24 offset)
+   </code>"
+  (defun logappn-fn (args)
+    (cond ((endp args) 0)
+          ((endp (cdr args)) 0)
```

Comment:
I think rather than returning 0 on an odd number of args, I'd prefer either an error (odd number of arguments where an even number was expected) or returning that last argument as the final tail. I think that would be useful since it would let this function produce a fully general nest of logapps, not only one ending in 0.